core/src/org/javarosa/xpath/XPathArityException.java

```diff
@@ -0,0 +1,14 @@
+package org.javarosa.xpath;
+
+/**
+ * Created by wpride1 on 3/28/15.
+ */
+public class XPathArityException extends XPathException {
+    public XPathArityException() {
+
+    }
+
+    public XPathArityException(String s, int required, int actual) {
+        super("cannot handle " + s + " had " + actual + " arguments but required " + required + ".");
```

Comment:
In order to make sure this is localize, it would likely be helpful to keep these parameters (function name, expected, provided), so we can catch the typed exception and throw a message with those details that's localized in the end platform (in addition to having this message for technical/fallback info)